### PR TITLE
ISPN-10403 Add queries and async-cache functional tests

### PR DESCRIPTION
**Pull request by gustavolira**
https://issues.jboss.org/browse/ISPN-10403

### CODE REVIEW

File: query/src/test/java/org/infinispan/query/test/Book.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.annotations.IndexedEmbedded;
+
+@Indexed
+public class Book {
```
Comment (karesti) [inline query/src/test/java/org/infinispan/query/test/Book.java]: should this file contain equals and hashcode as well ?

Comment (gustavolira) [reply query/src/test/java/org/infinispan/query/test/Book.java]: Done

**Comment by gustavolira** [comment]
> Is there a way to simplify the tests about the async write behind? they seem complex and prone to random failures

@karesti TBH this test has existed since 2015 on jdg-functional-tests project. I'm just moving this test to upstream

**Comment by ryanemerson** [comment]
Thanks @gustavolira

File: query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java [hunk 2]
```diff
@@ -46,8 +46,17 @@ public QueryPhrasesTest() {
       cleanup = CleanupPhase.AFTER_METHOD;
    }
 
+   @SuppressWarnings("unused")
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java]: What is the purpose of this unused constructor?

File: query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java [hunk 3]
```diff
@@ -46,8 +46,17 @@ public QueryPhrasesTest() {
...
+   public QueryPhrasesTest(EmbeddedCacheManager cacheManager) {
+      this.cacheManager = cacheManager;
+      cleanup = CleanupPhase.AFTER_METHOD;
+   }
+
    @Override
    protected EmbeddedCacheManager createCacheManager() throws Exception {
+      if (this.cacheManager != null) {
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java]: I smell a trick here...

**Comment by anistor** [comment]
I hope jdg-functional-tests does not depend on any of these classes. I'm tempted to remove the unused constructor.

File: query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java (hunk 2 above)
Comment (gustavolira) [reply query/src/test/java/org/infinispan/query/queries/phrases/QueryPhrasesTest.java]: @anistor I'm using this constructor without argument on jdg-functional-tests wrapping QueryPhrasesTest.java

**Pull request merged**